packages/functional-tests/pages/baseTokenCode.ts

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { BaseLayout } from './layout';
+import { expect } from '../lib/fixtures/standard';
+
+export class BaseTokenCodePage extends BaseLayout {
```

Comment:
Since many of the pages that submit a code are essentially the same, this is a common base class that any 'code' page can use.